### PR TITLE
{toolchain} foss/2019a (+ HPL 2.3 as test case)

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in this PR)
node2452.golett.os - Linux centos linux 7.5.1804, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/e6428d6cb771ba8cc902f1b70aed1589 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in this PR)
node3102.skitty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/9894e51860089989a83967596d47ef51 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in this PR)
node2094.delcatty.os - Linux centos linux 7.5.1804, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/868366c2244a4db11faef80d6c740a9b for a full test report.

**Comment by migueldiascosta** [comment]
Test report by @migueldiascosta
**SUCCESS**
Build succeeded for 17 out of 17 (6 easyconfigs in this PR)
grc-cluster1 - Linux centos 6.10, Intel(R) Xeon(R) CPU E5-2640 0 @ 2.50GHz, Python 2.7.14
See https://gist.github.com/5a7748e9b105a815bb9fb24ad8769f72 for a full test report.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**FAILED**
Build succeeded for 1 out of 7 (6 easyconfigs in this PR)
b-cn0315.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/d622a3c2171041cc043a7803eb48ff0a for a full test report.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**FAILED**
Build succeeded for 1 out of 7 (6 easyconfigs in this PR)
b-cn0315.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/6ef6c780ccc1ef59a2f1d5910c1521ce for a full test report.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 7 out of 7 (6 easyconfigs in this PR)
b-cn0315.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/409e0cb47163d1c8aad26f2a32caf38f for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in this PR)
nic27 - Linux centos linux 7.5.1804, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/379cde604b4d6083b6386b671354e396 for a full test report.

**Comment by boegel** [comment]
Detailed test report on re-building all easyconfigs that use `foss/2018b` as toolchain with the proposed `foss/2019a` instead is available at https://gist.github.com/boegel/68b542fa564fd370e8b0e7710a320f6f .

Compilation failed for a handful of easyconfigs with cryptic errors (to be figured out), but I don't think any of these should block us from going forward with the current `foss/2019a` definition (and this was not contested during the last EasyBuild conf call, see https://github.com/easybuilders/easybuild/wiki/Conference-call-notes-20190123#test-report-for-proposed-2019a-common-toolchains).

Summary: 234 successfully installed easyconfigs using `foss/2019a` (+ 102 more with `GCCcore/8.2.0` subtoolchain) vs 4+3 failed installations.

File: easybuild/easyconfigs/f/foss/foss-2019a.eb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+easyblock = 'Toolchain'
+
+name = 'foss'
+version = '2019a'
+
+homepage = '(none)'
```
Comment (ocaisa) [inline easybuild/easyconfigs/f/foss/foss-2019a.eb]: Could link to https://easybuild.readthedocs.io/en/develop/Common-toolchains.html#foss-toolchain here

File: easybuild/easyconfigs/f/foss/foss-2019a.eb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+blasver = '0.3.5'
+blas = '%s-%s' % (blaslib, blasver)
+
+# toolchain used to build foss dependencies
+comp_mpi_tc_name = 'gompi'
+comp_mpi_tc = (comp_mpi_tc_name, version)
+
+# we need GCC and OpenMPI as explicit dependencies instead of gompi toolchain
+# because of toolchain preparation functions
+# For binutils, stick to http://wiki.osdev.org/Cross-Compiler_Successful_Builds
```
Comment (ocaisa) [inline easybuild/easyconfigs/f/foss/foss-2019a.eb]: This comment is not relevant here

File: easybuild/easyconfigs/h/HPL/HPL-2.3-foss-2019a.eb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+source_urls = ['http://www.netlib.org/benchmark/%(namelower)s']
+sources = [SOURCELOWER_TAR_GZ]
+patches = ['HPL_parallel-make.patch']
+checksums = [
+    '32c5c17d22330e6f2337b681aded51637fb6008d3f0eb7c277b163fadd612830',  # hpl-2.3.tar.gz
+    '2a5bf9c4f328049828ddecec7ba3f05a9e25d236f4212747c53bd22fea80c5e6',  # HPL_parallel-make.patch
+]
+
+# fix Make dependencies, so parallel build also works
```
Comment (ocaisa) [inline easybuild/easyconfigs/h/HPL/HPL-2.3-foss-2019a.eb]: Comment not in relevant place

**Pull request merged**